src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,8 +1337,1089 @@ public boolean mapSwitchControlValue(String value) {
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case ENCRYPTED:
+				Boolean encrypted = mapSwitchControlValue(value);
+				removeUnusedStatsAndControlByEncrypted(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				createStream.setPassphraseSet(encrypted);
+
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case PASSPHRASE:
+				createStream.setPassphrase(value);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.PASSPHRASE.getName(), createStream.getPassphrase()));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case REJECT_UNENCRYPTED_CALLERS:
+				Boolean rejectUnencryptedCallers = mapSwitchControlValue(value);
+				createStream.setStrictMode(rejectUnencryptedCallers);
+				addAdvanceControlProperties(advancedControllableProperties, createSwitch(
+						stats, streamControllingGroup + StreamControllingMetric.REJECT_UNENCRYPTED_CALLERS.getName(), rejectUnencryptedCallers, DecoderConstant.DISABLE, DecoderConstant.ENABLE));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case CREATE:
+				performCreateStreamControl(createStream);
+				break;
+			case CANCEL:
+				createStream = defaultStream();
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			default:
+				throw new IllegalStateException("Unexpected value: " + controllableProperty);
+		}
+	}
+
+	/**
+	 * This method is used to perform create stream control
+	 *
+	 * @param streamInfo set of stream config info
+	 */
+	private StreamInfo performCreateStreamControl(StreamInfo streamInfo) {
+		StreamInfo streamInfoResult = new StreamInfo();
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				String request = streamInfo.jsonRequest();
+				if (logger.isDebugEnabled()) {
+					logger.debug(request);
+				}
+				System.out.println(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.STREAMS));
+
+				StreamDataWrapper streamDataWrapper = doPost(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.STREAMS), request, StreamDataWrapper.class);
+				if (streamDataWrapper == null) {
+					throw new ResourceNotReachableException(DecoderConstant.CREATE_CONTROL_ERR);
+				} else {
+					isCreatedStreamControl = true;
+				}
+			} else {
+				throw new ResourceNotReachableException(DecoderConstant.CREATE_CONTROL_ERR);
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(DecoderConstant.CREATE_CONTROL_ERR + e.getMessage());
```

Comment:
please add original exception as the 2nd argument of ResourceNotReachableException, not to lose the context

Reply:
I fixed it